### PR TITLE
feat(rafiki): add rafiki as a submodule and setup local instance

### PR DESCRIPTION
**Pull request by dragosp1011**
### Changes
- add rafiki as a submodule
- add rafiki services to dev docker compose

### CODE REVIEW

File: docker/dev/docker-compose.yml [hunk 1]
```diff
@@ -10,24 +10,30 @@ services:
     ports:
       - '5433:5432'
     restart: unless-stopped
+    networks:
+      - testnet
     volumes:
       - pg-data:/var/lib/postgresql/data
+      - ./dbinit.sql:/docker-entrypoint-initdb.d/init.sql
```
Comment (raducristianpopa) [inline docker/dev/docker-compose.yml:17]: Should we have two `postgres` images (one for the wallet and one for Rafiki)?

File: docker/dev/docker-compose.yml [hunk 2]
```diff
@@ -46,5 +54,117 @@ services:
...
+      dockerfile: ./packages/http-signature-utils/Dockerfile
+    restart: always
+    ports:
+      - '3042:3000'
+    environment:
+      KEY_FILE: /workspace/private-key.pem
+    volumes:
+      - ./private-key.pem:/workspace/private-key.pem
+    networks:
+      - testnet
```
Comment (raducristianpopa) [inline docker/dev/docker-compose.yml:158]: Do we need the signatures application? 🤔

https://github.com/interledger/rafiki/tree/main/packages/http-signature-utils#http-signature-utils-library

File: docker/dev/docker-compose.yml (hunk 1 above)
Comment (dragosp1011) [reply docker/dev/docker-compose.yml:17]: for dev I don't think we need to separate them

File: docker/dev/docker-compose.yml (hunk 2 above)
Comment (dragosp1011) [reply docker/dev/docker-compose.yml:158]: looks like no, we will add it back if we will need it in the future

File: docker/dev/dbinit.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
+CREATE USER backend WITH PASSWORD 'backend';
+CREATE DATABASE backend;
+ALTER DATABASE backend OWNER TO backend;
+
+CREATE USER auth WITH PASSWORD 'auth';
+CREATE DATABASE auth;
+ALTER DATABASE auth OWNER TO auth;
```
Comment (beniaminmunteanu) [inline docker/dev/dbinit.sql:7]: These are both used in rafiki?
Or is the backend you are refering to there our testnet backend?

Comment (dragosp1011) [reply docker/dev/dbinit.sql:7]: both are for rafiki services

Comment (beniaminmunteanu) [reply docker/dev/dbinit.sql:7]: got it

**Review by beniaminmunteanu — DISMISSED** [review]
LGTM

**Review by raducristianpopa — APPROVED** [review]

**Pull request merged**